### PR TITLE
Refactor to use package structure

### PR DESCRIPTION
**Pull request by CompeyDev**
To improve structure and eliminate the overhead of syncing an entire repository, pesde v0.5.0-rc.15 adds support for a `target.scripts` manifest key, which allows for a package to export scripts that are installed into the `.pesde/` directory, and can be accessed by `pesde run`. 

This PR restructures the project to be a normal pesde package that exports scripts for different sync tools. The build script located at `.lune/build.luau` generates packages for tools supported by the core library, as defined in the top-level `pesde.toml` using the `meta` manifest key.

The core library current includes functionality abstracting around sourcemap & sync config generators, organized at a per-tool basis. At the moment, Rojo is the only supported tool, but supporting other tools should be fairly simple, and is planned for.

### CODE REVIEW

File: .lune/lib/channel.luau [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+---     local value = recv()
+---     print("received value:", value)
+--- end)
+--- ```
+type Watch<T> = {
+    value: T?,
+    receivers: { thread },
+}
+
+--- Crates a new `Watch` channel, returning its send and receive handles.
```
Comment (lukadev-0) [inline .lune/lib/channel.luau]:
```suggestion
--- Creates a new `Watch` channel, returning its send and receive handles.

```

File: stylua.toml [hunk 2]
```diff
@@ -1,2 +1,10 @@
+line_endings = "Unix"
+quote_style = "AutoPreferDouble"
+indent_type = "Spaces"
```
Comment (daimond113) [inline stylua.toml:3]: Use tabs instead.
```suggestion
indent_type = "Tabs"
```

File: .lune/typecheck.luau [hunk 3]
```diff
@@ -0,0 +1,16 @@
+--> Run luau-lsp analysis to check for type errors
+
+local process = require("@lune/process")
+
+local CommandBuilder = require("./lib/exec")
+
+process.exit(
+	CommandBuilder.new("~/.rokit/bin/luau-lsp")
```
Comment (daimond113) [inline .lune/typecheck.luau]: We should use pesde managed tooling for this.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Pesde packages
+*_packages/
+pesde.lock
+
+# Generated scripts
+.pesde/*
+!.pesde/.gitkeep
```
Comment (daimond113) [inline .gitignore:7]: Why?

File: .pesde/.gitkeep
Comment (daimond113) [inline .pesde/.gitkeep:1]: Again, why?

File: .lune/fmt.luau [hunk 5]
```diff
@@ -0,0 +1,7 @@
+--> Run stylua to check for formatting errors
+
+local process = require("@lune/process")
+
+local CommandBuilder = require("./lib/exec")
+
+process.exit(CommandBuilder.new("stylua"):withArg("."):withArgs(process.args):withStdioStrategy("forward"):exec().code)
```
Comment (daimond113) [inline .lune/fmt.luau]: Stylua isn't in the manifest. Probably Rokit managed here too.

File: .lune/lib/manifest.luau [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    | "CC-BY-4.0"
+    | "CC-BY-SA-4.0"
+    | "BSL-1.0"
+    | "EPL-2.0"
+    | "AGPL-3.0"
+
+export type DependencySpecifier = ((
+    { name: string, version: string, index: string? }
+    | { workspace: string, version: string }
+    | { repo: string, rev: string, path: string? }
```
Comment (daimond113) [inline .lune/lib/manifest.luau]: Git specifiers don't have a target field.

**Review by daimond113 — CHANGES_REQUESTED** [review]

File: .lune/lib/manifest.luau [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+}) | { wally: string, version: string, index: string? }
+
+export type PackageTarget = {
+    environment: "luau" | "lune" | "roblox" | "roblox_server",
+    lib: string,
+} | ({ environment: "luau" | "lune" } & ({
+    bin: string,
+} | {
+    scripts: { [string]: string },
+}))
```
Comment (lukadev-0) [inline .lune/lib/manifest.luau:40]:
```suggestion
export type PackageTarget = {
    environment: "roblox" | "roblox_server",
    lib: string,
} | {
    environment: "luau" | "lune",
    lib: string,
    bin: string,
    scripts: { [string]: string },
}
```

File: .lune/typecheck.luau (hunk 3 above)
Comment (CompeyDev) [reply .lune/typecheck.luau]: This was for testing -- I missed updating this. Fixed in https://github.com/pesde-pkg/scripts/pull/3/commits/677e49473b6cc3835a19dbe636a1217102196541.

File: .gitignore (hunk 4 above)
Comment (CompeyDev) [reply .gitignore:7]: We exclude the subprojects in the `.pesde` directory, but keep the directory. I plan on placing a README there to explain what happens to the directory.

File: .lune/fmt.luau (hunk 5 above)
Comment (CompeyDev) [reply .lune/fmt.luau]: Addressed in https://github.com/pesde-pkg/scripts/pull/3/commits/5e0905673164aeedb025027d7fc484a7438111b6.

**Review by lukadev-0 — CHANGES_REQUESTED** [review]

File: .lune/lib/manifest.luau [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    | "EPL-2.0"
+    | "AGPL-3.0"
+
+export type DependencySpecifier = ((
+    { name: string, version: string, index: string? }
+    | { workspace: string, version: string }
+    | { repo: string, rev: string, path: string? }
+) & {
+    target: string?,
+}) | { wally: string, version: string, index: string? }
```
Comment (lukadev-0) [inline .lune/lib/manifest.luau]:
```suggestion
export type DependencySpecifier = ((
    { name: string, version: string, index: string? }
    | { workspace: string, version: string }
) & {
    target: string?,
})
    | { wally: string, version: string, index: string? }
    | { repo: string, rev: string, path: string? }
```

File: stylua.toml (hunk 2 above)
Comment (CompeyDev) [reply stylua.toml:3]: No. :)

Comment (daimond113) [reply stylua.toml:3]: Yes. :)

**Review by lukadev-0 — APPROVED** [review]
LGTM!

**Review by daimond113 — APPROVED** [review]
LGTM!

**Pull request merged**